### ISSUE TITLE
Syncing fails if IPv6 is disabled. Gives a non-descriptive error.

### ISSUE THREAD

**Issue by clay53** [ankitects/anki#1758]
I had disabled IPv6 in my kernel (`systemd-boot`) with `ipv6.disable=1` but I brought it back when I found another workaround to the problems I was having with it when I got it working on my laptop. Realized it was still disabled when I compared the out put of resolvectl to that on my laptop and saw that IPv6 was shown on my laptop but not my desktop. Still weird because Anki used to work with this disabled (same version). Maybe removing disables on other parts of my system allows this error to occur or updates to the kernel. Disabling the block in my kernel now allows me to ping ankiweb.net, sync on Anki, and IPv6 now shows on my interface on resolvectl. Good to note that accessing ankiweb.net through the browser always worked.

Was writing a report on the forums for this when I fixed it that has more details (the error I was getting is included):
When I try to sync, I get the error:
```
Connection timed out. Please try again. If you see frequent timeouts, please try a different network connection.

Error details: error sending request for url (): error trying to connect: operation timed out
```

When it first tries syncing, it's saying "Processing" with a full blue bar until it shows the error above on screen.

Here are the full logs (running from build: `ANKIDEV=1 LOGTERM=1 TRACESQL=1 ./run`):
```
INFO: Analyzed target //qt:runanki (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //qt:runanki up-to-date:
  .bazel/bin/qt/runanki
INFO: Elapsed time: 0.301s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
Running with temporary Qt5 compatibility shims.
Run with DISABLE_QT5_COMPAT=1 to confirm compatibility with Qt6.
Preparing to run...
Serving on http://127.0.0.1:37353
Qt info:

GLImplementation: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.6
Using Default SG Backend: yes
Using Software Dynamic GL: no
Using Angle: no

Init Parameters:
  *  application-name Anki
  *  browser-subprocess-path /home/clay53/.cache/bazel/_bazel_clay53/71a6a62a42e8918c51d48afc3ba39bda/execroot/ankidesktop/bazel-out/k8-fastbuild/bin/qt/runanki.runfiles/pyqt6/PyQt6/Qt6/libexec/QtWebEngineProcess
  *  create-default-gl-context
  *  disable-features DnsOverHttpsUpgrade,ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture
  *  disable-setuid-sandbox
  *  disable-speech-api
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess
  *  enable-threaded-compositing
  *  gpu-preferences UAAAAAAAAAAoAAAQAAAAAAAAAAAAAAAAAABgAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAgAAAAAAAAACAAAAAAAAAA=
  *  in-process-gpu
  *  use-gl desktop

Starting main loop...
1648596355.380 GET /_anki/legacyPageData
1648596355.412 GET /_anki/css/webview.css
1648596355.413 GET /_anki/css/toolbar.css
1648596355.413 GET /_anki/js/webview.js
1648596355.416 GET /_anki/js/vendor/jquery.min.js
1648596355.416 GET /_anki/js/toolbar.js
1648596355.419 GET /_anki/imgs/refresh.svg
sql: PRAGMA locking_mode='exclusive'
sql: PRAGMA page_size=4096
sql: PRAGMA cache_size=-40960
sql: PRAGMA legacy_file_format=0
sql: PRAGMA journal_mode='wal'
sql: select null from sqlite_master where type = 'table' and name = 'col'
sql: select ver from col
sql: select mod from col
sql: begin exclusive
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'sched2021'
sql: SELECT val FROM config WHERE KEY = 'interruptAudioWhenAnswering'
sql: select id, name from decks order by name
sql: SELECT id,   name,   mtime_secs,   usn,   common,   kind FROM decks
sql: select null from cards where did=1
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'localOffset'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'rollover'
sql: select crt from col
sql: SELECT val FROM config WHERE KEY = 'creationOffset'
sql: SELECT val FROM config WHERE KEY = 'lastUnburied'
sql: SELECT val FROM config WHERE KEY = 'normalize_note_text'
sql: DROP TABLE IF EXISTS search_cids;
sql: CREATE TEMPORARY TABLE search_cids (cid integer PRIMARY KEY NOT NULL);
sql: insert into search_cids select c.id from cards c where (c.queue in (-2,-3))
sql: SELECT id,   nid,   did,   ord,   cast(mod AS integer),   usn,   type,   queue,   due,   cast(ivl AS integer),   factor,   reps,   lapses,   left,   odue,   odid,   flags,   data FROM cards where id in (select cid from search_cids)
sql: drop table if exists search_cids
sql: SELECT val,   usn,   mtime_secs FROM config WHERE KEY = 'lastUnburied'
sql: INSERT   OR REPLACE INTO config (KEY, usn, mtime_secs, val) VALUES ('lastUnburied', -1, 1648596362, x'31')
sql: SELECT val FROM config WHERE KEY = 'localOffset'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'rollover'
sql: select crt from col
sql: SELECT val FROM config WHERE KEY = 'creationOffset'
sql: SELECT val FROM config WHERE KEY = 'collapseTime'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'sched2021'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT did,   -- new   sum(queue = 0),   -- reviews   sum(     queue = 2     AND due <= 1   ),   -- interday learning   sum(     queue = 3     AND due <= 1   ),   -- intraday learning   sum(     (       CASE         2         WHEN 2 THEN (           -- v2 scheduler           (             queue = 1             AND due < 1648597562           )           OR (             queue = 4             AND due <= 1648597562           )         )         ELSE (           -- v1 scheduler           CASE             WHEN queue = 1             AND due < 1648597562 THEN left / 1000             ELSE 0           END         )       END     )   ),   -- total   COUNT(1) FROM cards group by did
sql: SELECT id,   name,   mtime_secs,   usn,   config FROM deck_config
sql: SELECT val FROM config WHERE KEY = 'curDeck'
sql: SELECT COUNT(),   coalesce(sum(time) / 1000.0, 0.0) FROM revlog WHERE id > 1648540800000   AND type != 4
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: select mod, scm, ls from col
sql: select id, name from decks order by name
sql: SELECT id,   name,   mtime_secs,   usn,   common,   kind FROM decks
sql: select null from cards where did=1
sql: SELECT val FROM config WHERE KEY = 'lastUnburied'
sql: SELECT val FROM config WHERE KEY = 'localOffset'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'rollover'
sql: select crt from col
sql: SELECT val FROM config WHERE KEY = 'creationOffset'
sql: SELECT val FROM config WHERE KEY = 'collapseTime'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT val FROM config WHERE KEY = 'sched2021'
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: SELECT did,   -- new   sum(queue = 0),   -- reviews   sum(     queue = 2     AND due <= 1   ),   -- interday learning   sum(     queue = 3     AND due <= 1   ),   -- intraday learning   sum(     (       CASE         2         WHEN 2 THEN (           -- v2 scheduler           (             queue = 1             AND due < 1648597562           )           OR (             queue = 4             AND due <= 1648597562           )         )         ELSE (           -- v1 scheduler           CASE             WHEN queue = 1             AND due < 1648597562 THEN left / 1000             ELSE 0           END         )       END     )   ),   -- total   COUNT(1) FROM cards group by did
sql: SELECT id,   name,   mtime_secs,   usn,   config FROM deck_config
sql: SELECT val FROM config WHERE KEY = 'curDeck'
sql: SELECT COUNT(),   coalesce(sum(time) / 1000.0, 0.0) FROM revlog WHERE id > 1648540800000   AND type != 4
sql: SELECT val FROM config WHERE KEY = 'schedVer'
sql: select mod from col
sql: PRAGMA page_size=4096
sql: PRAGMA legacy_file_format=0
sql: rollback
sql: PRAGMA journal_mode='wal'
sql: select null from sqlite_master where type = 'table' and name = 'media'
sql: begin immediate
sql: select dirMod, lastUsn from meta
sql: commit
sql: select dirMod, lastUsn from meta
1648596362.876 GET /_anki/legacyPageData
1648596362.895 GET /_anki/legacyPageData
1648596362.905 GET /_anki/css/webview.css
1648596362.906 GET /_anki/css/deckbrowser.css
1648596362.906 GET /_anki/js/webview.js
1648596362.908 GET /_anki/js/vendor/jquery.min.js
1648596362.908 GET /_anki/js/vendor/jquery-ui.min.js
1648596362.909 GET /_anki/js/deckbrowser.js
1648596362.915 GET /_anki/css/webview.css
1648596362.917 GET /_anki/css/toolbar.css
1648596362.918 GET /_anki/css/toolbar-bottom.css
1648596362.919 GET /_anki/js/vendor/jquery.min.js
1648596362.919 GET /_anki/js/webview.js
1648596362.933 GET /_anki/imgs/gears.svg
sql: select mod from col
sql: begin exclusive
```

Fixes I've tried:
* different versions
  * old version (I think 2.1.40)
    * Previously worked on this version when it was new, months ago
  * `anki` from AUR
  * `anki-bin` from AUR
  * `anki-official-binary-bundle` from AUR
  * latest official version from website `anki-2.1.49-linux`
  * installing from flatpak
  * building from git (`7977a0dc1fb8f6e0de0648fc7b24a7bb18bb0cc0`)
* deleting all add-ons
* going through install setup and reversing it manually to ensure all data deleted before installing again
* deleting user data from ~/.local/shared
* trying to diagnose through wireshark (didn't really help)
* trying to diagnose using netstat (seems to be trying to connect to another local address)
* connecting through NordVPN
* am able to access ankiweb.net
* doing a full system update
* restarting
* installing Anki on different of OS of same computer works
* laptop with same OS/network setup except ethernet swapped with WIFI works

I am not running any proxies or VPNs. My network setup is systemd/networkd/resolved. A laptop and android device is able to sync without issue. Ankiweb shows all sync data as expected. Using a VPN does not seem to have any effect.

Just noticed pinging of ankiweb.net does not actually work (works on laptop) despite the website being accessible.

preforming `dig ankiweb.net` succeeds
```
; <<>> DiG 9.18.1 <<>> ankiweb.net
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 45494
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;ankiweb.net.			IN	A

;; ANSWER SECTION:
ankiweb.net.		300	IN	A	142.132.136.175
ankiweb.net.		300	IN	A	142.132.136.176
ankiweb.net.		300	IN	A	162.55.238.74
ankiweb.net.		300	IN	A	162.55.238.75

;; Query time: 23 msec
;; SERVER: 192.168.1.1#53(192.168.1.1) (UDP)
;; WHEN: Tue Mar 29 19:45:19 EDT 2022
;; MSG SIZE  rcvd: 104
```

Other system info (working now):
```
OS: Arch Linux x86_64
Host: Z170X-Gaming 3
Kernel: 5.16.16-arch1-1
Uptime: 30 mins
Packages: 1568 (pacman), 4 (flatpak)
Shell: bash 5.1.16
Resolution: 1920x1080, 1920x1080, 2560x1440
WM: LeftWM
Theme: Adwaita [GTK2/3]
Icons: Adwaita [GTK2/3]
Terminal: alacritty
CPU: Intel i7-6700 (8) @ 4.000GHz
GPU: NVIDIA GeForce RTX 2060 Rev. A
Memory: 3150MiB / 32058MiB
```

Forgot to note: doing `ping -v ankiweb.net` gave some thing about socket protocol not being supported which helped clue me in. Thought it was weird though because dig ankiweb.net only gives IPv4 addresses. Large services like Anki should also always support IPv4 only devices/networks.

**Comment by dae** [comment]
ankiweb.net is IPv4 only. The fact that you can't ping it outside of Anki seems to point to some issue with your network configuration. Do you get the same results when you attempt to ping the ip addresses instead of the DNS name?

**Comment by clay53** [comment]
> ankiweb.net is IPv4 only. The fact that you can't ping it outside of Anki seems to point to some issue with your network configuration. Do you get the same results when you attempt to ping the ip addresses instead of the DNS name?

Sorry for the long time to reply. I tried setting the parameter on my kernel again and clearing my DNS cache and couldn't reproduce it. Not sure what was happening at the time. Removing the kernel parameter was the only thing I changed between tests so it definitely fixed whatever was happening but that _may_ have not been an Anki issue. Otherwise, it would be nice if Anki gave what connection failed instead of just saying `()` (probably shows an actual url for other things but maybe there are some cases where it fails).
